packages/flutter/lib/src/widget_previews/widget_previews.dart

```diff
@@ -183,17 +219,122 @@ final class Preview {
 /// ```
 /// {@end-tool}
 ///
-/// **Important Note:** all values provided to the `Preview()` instances included in the
-/// `previews` list must be constant and non-private.
+/// **Important Note:** all values provided to [MultiPreview] derived annotations must
+/// be constant, and callback parameters must also be static and non-private. These
+/// restrictions do not apply when creating instances of [MultiPreview] at runtime, including
+/// within [Preview.transform()].
+///
+/// See also:
+///   - [Preview], the annotation used to mark functions that return a widget preview.
+///   - [flutter.dev/to/widget-previews](https://flutter.dev/to/widget-previews)
+///     for details on getting started with widget previews.
 abstract base class MultiPreview {
   /// Creates a [MultiPreview] annotation instance.
   const MultiPreview();
 
   /// The set of [Preview]s to be created for the annotated function.
-  ///
-  /// **Important Note:** this getter must be overridden with a field, not a getter, otherwise
-  /// the list of [Preview]s will not be detected.
   List<Preview> get previews;
+
+  /// Applies a transformation to each [Preview] in [previews].
+  ///
+  /// Overriding this method allows for [MultiPreview] implementations to initialize more
+  /// complicated previews that would not otherwise be possible due to restrictions around constant
+  /// constructors used for annotations.
+  ///
+  /// See also:
+  ///   - [PreviewBuilder], a utility for building and modifying [Preview]s.
+  @mustCallSuper
+  List<Preview> transform() => previews.map((Preview e) => e.transform()).toList();
+}
+
+/// A utility class used to build a [Preview] instance.
+final class PreviewBuilder {
+  /// Creates a [PreviewBuilder] with no initial state.
+  PreviewBuilder();
+
+  /// Creates a [PreviewBuilder] initialized with the values set in [preview].
+  PreviewBuilder.fromPreview(Preview preview)
+    : group = preview.group,
+      name = preview.name,
+      size = preview.size,
+      textScaleFactor = preview.textScaleFactor,
+      wrapper = preview.wrapper,
+      theme = preview.theme,
+      brightness = preview.brightness,
+      localizations = preview.localizations;
+
+  /// The group the preview belongs to.
+  ///
+  /// Previews with the same group name will be displayed together in the
+  /// previewer. If not provided, 'Default' is used.
+  String? group;
```

Comment:
I'm not sure if we have dartdoc best practices/suggestions around this, but I wonder if instead all these members could have comments like `/// See [Preview.group]` so that we don't have to keep the documentation updated in two places.

Reply:
I totally forgot about that! Thanks for reminding me :)